src/service/resources-impl.js

```diff
@@ -435,6 +435,53 @@ export class Resources {
   }
 
   /**
+   * Runs the specified mutation on the element and ensures that measures
+   * and layouts performed for the affected elements.
+   * @param {!Element} element
+   * @param {function()} mutator
+   * @return {!Promise}
+   */
+  mutateElement(element, mutator) {
+    const calcRelayoutTop = () => {
+      const box = this.viewport_.getLayoutRect(element);
+      if (box.width != 0 && box.height != 0) {
+        return box.top;
+      }
+      return -1;
```

Comment:
width=0 and height=0 means that the element is invisible. We can't trust `top` in this case and generally have nothing to remeasure. One example, on Chrome, if you set `display:none` on an element, it will continue to report the "old" top before it was invisible no matter what happens to the DOM after.